src/App.js

```diff
@@ -32,14 +32,40 @@ class App extends Component {
             path: '/game'
         });
 
+
+        // Player messages
         this.io.on('connect', () => {
-            this.io.on('msg', msg => console.log(msg));
-            this.io.emit('msg', 'hello');
+            this.io.emit('join game', this.state.currentPlayer);
+        });
+
+        this.io.on('set player id', id => {
+            console.log(`You are player ${id}`);
+            this.setState({ currentPlayer: {
+                ...this.state.currentPlayer,
+                id: id
+            }
+            });
+        });
+
+
+        // Server messages
+        this.io.on('player joined', playerState => {
+            console.log(`Player ${playerState.id} joined the game.`, playerState);
+        });
+
+        this.io.on('player list update', playersList => {
+            console.log('Updated player list.', playersList);
+            this.setState({ otherPlayers: playersList.filter(player => player.id !== this.state.currentPlayer.id) });
```

Comment:
We can avoid this complexity by having the state shaped more like this:

```js
type ID = string;

type State = {
  me: ID // just the player id, altho maybe this is not the ideal key name
  players: {
    [key: ID]: Player
  }
}
```

Let me know if this notation doesn't make sense.